platforms/software/dependency-management/src/main/java/org/gradle/internal/component/ConfigurationNotConsumableException.java

```diff
@@ -14,9 +14,14 @@
  * limitations under the License.
  */
 
-package org.gradle.internal.exceptions;
+package org.gradle.internal.component;
 
-public class ConfigurationNotConsumableException extends IllegalArgumentException {
+/**
+ * This exception is thrown during variant aware dependency resolution
+ * when a specific configuration of a dependency is requested by name
```

Comment:
🤔 `configuration of a dependency` sounds weird to me. Users use a dependency to request a configuration of a component. They are not requesting a configuration of the dependency

Reply:
Yay for overloaded terms! 🤮

How about "This exception is thrown during variant aware dependency resolution when a dependency explicitly requests a specific configuration of a component by name."

Reply:
Works for me